lib/Service/PageService.php

```diff
@@ -1084,4 +1096,12 @@ public function getBacklinks(int $collectiveId, int $id, string $userId): array
 
 		return $backlinks;
 	}
+
+	private function generateSlugForPage(string $title, ?File $file): ?string {
+		if (!$file) {
+			return null;
+		}
+
+		return $this->slugGeneratorService->generatePageSlug($title);
```

Comment:
I guess that you wanted to pass the `$fileId` here and add it to the sluggified title in `generatePageSlug`?

Reply:
This still seems broken to me. The backend code doesn't add the page fileId to the slug, but the frontend code expects it to be there.

Reply:
no, it's not needed. We're matching `pageId` and `pageSlug` separately on FE. This needed because slug can change and we should redirect user to a new slug based on `pageId`